src/shave.cpp

```diff
@@ -350,7 +351,26 @@ void filterFeatures(vtzero::tile_builder* finalvt,
                 feature_builder.set_id(feature.id());
             }
             feature_builder.set_geometry(feature.geometry());
+
+            std::string property_filter_type = property_filter.type;
+            std::vector<std::string> properties = property_filter.values;
+            auto keytable = layer.key_table();
```

Comment:
This should be:

```
            std::string const& property_filter_type = property_filter.type;
            std::vector<std::string> const& properties = property_filter.values;
            auto const& keytable = layer.key_table();
```

To avoid copies (potentially expensive) being made of those objects. In short, in Javascript every variable is a reference. In C++ you need to explicitly ask for a reference. The `&` means it is a reference, and therefore not copied.

Reply:
👍 as same as the about one

Reply:
👍 gotcha, & means we only pass the reference to the next step, not pass an copy of it.

Reply:
👍